### ISSUE TITLE
Add rounding at the bottom of the window

### ISSUE THREAD

**Issue by Zbergen-cli** [Zonnev/elementaryos-firefox-theme#144]
Hello.

In EOS Odin, all native applications have a rounding at the top and bottom of the window. It's worth adding this to the theme for better integration.

**Comment by Zonnev** [comment]
Hi thanks for your feedback. I wish I could add rounded corners to Firefox just like the native windows in eOS have now but I don't have the knowledge to implement it. I think when eOS would see Firefox as a native app, things would be much easier ;) Although... even the native browser Web doesn't have all 4 corners rounded!

Search for "rounded" in [this article](https://blog.elementary.io/look-and-feel-changes-elementary-os-6/) and you will read a little about developing the rounded corners by the devs of eOS Odin.  

So, yes I think too that it's definitely worth adding this but since Firefox devs and eOS devs aren't working together, I think Firefox will always look a little off... 

**Added:** I even worry with every new version of Firefox that the rounded top corners will stay there or not, because even these existing corners look terrible to me! :(

**Comment by Zbergen-cli** [comment]
I found some ancient references to this problem. I think it's worth reminding the developers about it, which is what I'll do now. I'll let them know if I can get through to them, but in the meantime I'll close the thread.

**Comment by ChildishGiant** [comment]
To my knowledge rounding Firefox corners would require forking and patching it, not just some CSS. As Elementary don't ship Firefox I highly doubt they themselves will ever do this.

**Comment by Zbergen-cli** [comment]
For this reason, we contacted [mozilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1706752) directly. Fingers crossed that they fix it.

I was reminded of how Canonical supported global menu patches in Unity for many applications. Perhaps a similar fate awaits EOS for rounding in other applications.

**Comment by Zonnev** [comment]
@Zbergen-cli Maybe open a new bug at Mozilla for the rounded corners in eOS Odin? 

The rounded corners in eOS Odin are written with the help of a new library called [libhandy](https://www.reddit.com/r/elementaryos/comments/nb46c0/why_the_web_apps_corners_are_not_rounded/gxxnnp1?utm_source=share&utm_medium=web2x&context=3). Knowing it is now part of the Gnome app development platform and I think when it is used in more Linux distributions, it would have a chance Firefox developers will use it too. But for now I think only eOS Odin use it, so the chances are low, but I'm not sure about this.

**Comment by Zonnev** [comment]
Please keep this issue open so everyone will notice the lack of rounded corners in Firefox and follow or participate in the development towards a solution.

**Comment by Zbergen-cli** [comment]
@Zonnev, I thought about opening up a new problem, but to be honest I was hesitant to do so and posted to an existing problem.

**Comment by Zonnev** [comment]
I just watched a video about Gnome 41 and it has rounded bottom borders as eOS 6. This has to do with the successor of libhandy, libadwaita which does draw the 4 rounded corners in Gnome 41. This looks hopefull for us, since future Gnome and Firefox users will want 4 rounded corners on a Firefox window as well. I think soon I will try to install the new Gnome to see how Firefox looks in it. If it looks terrible, I will try to report this bug at Mozilla.

**Comment by Zonnev** [comment]
![Schermafdruk van 2021-09-24 15 54 49](https://user-images.githubusercontent.com/32688765/134686603-bdf204a5-ae6e-47c2-8ea5-916f58ed32e0.png)
Firefox looks okay on Gnome 41 but is **not** integrated with the new libadwaita making 4 rounded window corners. I guess I can ask the devs to look at it in the near future when Gnome 41 is more common. 

By the way, notice how Gnome Boxes look on eOS Odin, it has square window corners! :(

**Comment by Zonnev** [comment]
There is talk about rounded bottom corners here: https://bugzilla.mozilla.org/show_bug.cgi?id=1509931#c31

**Comment by Zonnev** [comment]
It's reported that the 'ugly' rounded window corners on Linux and the missing rounded corners for the other theme modes dark, light and Alpenglow, will be fixed in Firefox 96 which will come to us in January 2022. 

I think the rounded bottom borders will not make it this time but perhaps a little more complaining will be enough to make them work on that too. But first let's wait what they did when Firefox 96 comes out.

I will watch development on Nightly as well to report the changes made.

**Comment by Zonnev** [comment]
I have the latest Firefox Nightly (96) running with the new CSS, styling the rounded corners, but the rounded corners are not there! I tested Nightly on Ubuntu 21.10 and the rounded corners are working there for all theme modes.

I'm very disappointed.... It seems that elementary OS Odin is rendering Firefox squared because Firefox can not trigger it's rounded window corners in Odin because it has a different window management system than other Linux distro's...

**Comment by Zonnev** [comment]
I reported a bug about this: https://bugzilla.mozilla.org/show_bug.cgi?id=1744496

**Comment by Zonnev** [comment]
Yay! At Bugzilla my bug has been fixed for Firefox 97, this means that in February we see the final result of how the rounded corners look. Only the upper rounded corners are fixed.

I will test this bug fix in Nightly when it's released.

**Comment by Zonnev** [comment]
The bug fix works well, on top two beautiful rounded corners in latest Nightly.
![Schermafdruk van 2021-12-13 00 03 03](https://user-images.githubusercontent.com/32688765/145733204-e6e05c1d-11ce-42ab-9805-7b5babccdc8f.png)

**Comment by Zbergen-cli** [comment]
Waiting for February)

**Comment by Zbergen-cli** [comment]
After the release of Firefox 97, this problem is no longer relevant :) I think it can be closed.

**Comment by Zonnev** [comment]
Yes, the top window corners are rounded now but the bottom corners aren't made rounded yet. It seems eOS does not make the bottom corners round on non eOS apps. If so, I'm afraid Firefox devs can not do anything. If eOS makes rounded bottom corners for all apps, then it would be easy for Firefox devs to make these corners.

So yes, I will close this issue since there is nothing to do about rounded bottom corners so far. Maybe in the future it will be an eOS feature, who knows...

**Comment by Zonnev** [comment]
Maybe it's already well known by Firefox users on eOS, but I just found out you can get rounded bottom corners by enabling `widget.gtk.rounded-bottom-corners.enabled` in `about:config` ;)
